src/Features/CSharp/Portable/ValidateFormatString/ValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ValidateFormatString;
+
+namespace Microsoft.CodeAnalysis.CSharp.ValidateFormatString
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class ValidateFormatStringDiagnosticAnalyzer : AbstractValidateFormatStringDiagnosticAnalyzer
+    {
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpr = (InvocationExpressionSyntax)context.Node;
+            if (invocationExpr.ArgumentList?.Arguments == null)
+            {
+                return;
+            }
+
+            SimpleNameSyntax name;
+            // When calling string.Format(...), Format will be of type MemberAccessExpressionSyntax 
```

Comment:
Also, technically the whole `string.Format` is the member access expression, and `Format` is the Name.